bodo/pandas/physical/expression.cpp

```diff
@@ -519,6 +519,11 @@ std::shared_ptr<ExprResult> PhysicalUDFExpression::ProcessBatch(
     // Actually run the UDF.
     std::shared_ptr<table_info> udf_output;
     if (cfunc_ptr) {
+        if (cfunc_ptr == (table_udf_t)1) {
+            PyThreadState* save = PyEval_SaveThread();
```

Comment:
Just curious, are these `PyEval_SaveThread` calls fundamentally necessary? Seems good for safety but looking at the docs they don't seem necessary.

Reply:
Ok, makes sense.